
# HG changeset patch
# User Sam Lantinga <slouken@libsdl.org>
# Date 1297938710 28800
# Node ID d898ee5431f52dd630e311b325eb5a42c8607a60
# Parent  6bb01861c4c0d28dcea67e04ef723feaaf730d07
Fixed bug #1090 (SDL_BlitCopyOverlap() assumes memcpy() operates in order)

The much more complete fix is in SDL 1.3, but this is a band-aid that will fix the bug for 1.2.

diff -r 6bb01861c4c0 -r d898ee5431f5 src/video/SDL_blit.c
--- a/src/video/SDL_blit.c	Wed Feb 16 04:49:07 2011 -0800
+++ b/src/video/SDL_blit.c	Thu Feb 17 02:31:50 2011 -0800
@@ -214,7 +214,7 @@
 	dstskip = w+info->d_skip;
 	if ( dst < src ) {
 		while ( h-- ) {
-			SDL_memcpy(dst, src, w);
+			SDL_memmove(dst, src, w);
 			src += srcskip;
 			dst += dstskip;
 		}

